elab_bridge/Extractor.py

```diff
@@ -0,0 +1,371 @@
+        Data['elabftw']['extra_fields_groups'] = sorted(Data['elabftw']['extra_fields_groups'],
+                                                        key=lambda x: x['id'])
+        logger.debug(f"Reorganized group fields: {Data['elabftw']['extra_fields_groups']}")
+
+        # Reorganize fields
+        for field in Data.get('extra_fields', {}).values():
+            if field['group_id'] in Dict_group_field_map:
+                field['group_id'] = Dict_group_field_map[field['group_id']]
+        logger.debug(f"Reorganized fields: {Data.get('extra_fields')}")
+
+        # Save the updated JSON
+        savefile(output_jsonfile, Data)
+        logger.info(f"Updated JSON saved to {output_jsonfile}")
+        return Data
+
+    except AssertionError as e:
+        logger.error(f"AssertionError: {str(e)}")
+    except KeyError as e:
+        logger.error(f"KeyError: {str(e)}")
+    except ValueError as e:
+        logger.error(f"ValueError: {str(e)}")
+    except Exception as e:
+        logger.error(f"An unexpected error occurred: {str(e)}")
+    return None
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(
+        description=(
+            "This script provides tools to manipulate JSON files, including:\n"
+            "- Extracting specific fields from a JSON file and creating a new structured JSON.\n"
+            "- Merging two JSON files based on group field IDs: Extracting a group field from one "
+            "JSON file and appending it to another.\n"
+            "- Ordering fields within a JSON file by specified group field names."
+        ),
+        epilog=(
+            "Examples of usage:\n"
+            "  Extract fields:\n"
+            "python Extractor.py extract --jsonfile_extract input.json --id 1 --new_id 2 "
+            "--json_output output.json\n\n"
+
+            "  Merge JSON files:\n"
+            "python Extractor.py merge --jsonfiletocompleted file1.json --jsonfiletoextract "
+            "file2.json --id 3 --new_id 4 --json_completed merged.json\n\n"
+            "  Order JSON fields:\n"
+            "python Extractor.py order --jsonfile input.json --list_group_field_name field1 "
+            "field2 --output_jsonfile ordered.json\n\n"
+            "Options:\n"
+            "  extract   Extracts fields from a JSON file and saves the output.\n"
+            "  merge     Complete a JSON file by adding a group field from another JSON file.\n"
+            "  order     Orders fields in a JSON file by the specified group field names."
+        ),
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "operation",
+        choices=["extract", "merge", "order"],
+        help="Specify the operation to perform: 'extract' to extract fields or 'merge' to combine "
+             "files.",
+    )
+
+    # Arguments for extracting
+    parser.add_argument("--jsonfile_extract", help="The JSON file path for extraction.", type=str)
```

Comment:
same as with the Merger, there should be a version of the options with only one dash and on letter (for example: -j)